components/classes/StudentListPrintout.tsx

```diff
@@ -24,6 +24,8 @@ export type OptionsType = {
     | "shirtSize"
     | "pantsSize"
     | "elective"
+    | "noElective"
+    | "hasAllergies"
   )[];

```

Comment:
I personally wouldn’t do this. Columns and filters are visually separated on the UI and so should they be in the code. It’s not wrong per se, just slightly confusing.

I would separate out `columns` and `filters`, like this:
```ts
export type OptionsType = {
  language: LangCode;
  columns: (
    | "classNo"
    | "studentID"
    | "prefix"
    | "fullName"
    | "nickname"
    | "allergies"
    | "shirtSize"
    | "pantsSize"
    | "elective"
  );
  filters: (
    | "noElective"
    | "hasAllergies"
  )[];
  numEmpty: number;
  enableNotes: boolean;
  enableTimestamp: boolean;
};
```

Reply:
Great suggestion for future usage and expansion, edited.